src/main/java/clueless/Game.java

```diff
@@ -213,12 +204,26 @@ public boolean hasPlayer() {
 	 */
 	public Player whoHasClue(List<Card> suggestion) {
 		Player pl = null;
-
-		// TODO: complete logic
-
-		return pl;
+		
+		while(pl == null)
+		{
+			for (int i = 0; i < CHARACTER_TURN_ORDER.length; i++) 
+			{
+				if (this.characterMap.get(CHARACTER_TURN_ORDER[i]) instanceof Player 
+						&& Player.hasClue(suggestion)) 
+					{
+						pl = (Player) this.characterMap.get(CHARACTER_TURN_ORDER[i]);
```

Comment:
as mentioned in above comment, prob just return `(Player) this.characterMap.get(CHARACTER_TURN_ORDER[i])` directly here.